builtin/credential/okta/cli.go

```diff
@@ -48,6 +51,20 @@ func (h *CLIHandler) Auth(c *api.Client, m map[string]string) (*api.Secret, erro
 		data["provider"] = provider
 	}
 
+	nonce := base62.MustRandom(20)
+	data["nonce"] = nonce
+
+	go func() {
+		for {
+			resp, _ := c.Logical().Read(fmt.Sprintf("auth/%s/verify/%s", mount, nonce))
```

Comment:
What happens if the user responds with an incorrect number, or forgets about it? Will this loop forever?

Reply:
The call to b.Login would unblock (failed auth), and the login would clean up the cached value through the defer call [in here](https://github.com/hashicorp/vault/pull/15361/files#diff-8ae1cf8247bed192b76102070c547ceda736bdad7243c035ef829add132c9205R88). The next verify call in this loop will return a nil response and we return from this goroutine.

Reply:
Hmm. But if the `verify` call continues to return a `nil` response, won't this loop forever?

Reply:
You're right, I got things the other way around. The goroutine returns when we've received the number challenge from Okta, but before the user taps on that number on their device. For the case where the response is nil (no e.g. number challenge), this goroutine will run indefinitely until the program exits, i.e. the CLI finishes with either successful or failed auth.

What blocks during the login call is the login call on L68, so if the taps on the wrong number challenge, the request ends with failed auth pretty instantaneously:
```
Password (will be hidden):
In Okta Verify, tap the number '83'
Error authenticating: Error making API request.

URL: PUT http://localhost:8200/v1/auth/okta/login/foo@example.com
Code: 400. Errors:

* multi-factor authentication denied
```

Since this is not running on the server side, it will get GC'ed when the program exits but it's not great. I'll refactor this to make it cleaner.

Reply:
Cleaned up the logic a bit so that we return from this goroutine whenever the we're done with `Auth`. Going to mark as resolved but feel free to re-open if you've got further comments!